File: 3056-determine-if-a-cell-is-reachable-at-a-given-time/3056-determine-if-a-cell-is-reachable-at-a-given-time.cpp
class Solution {
public:
    bool isReachableAtTime(int sx, int sy, int fx, int fy, int t) {
        if(sx==fx and sy==fy and t==1){
            return false;
        }
        int x_diff = abs(fx-sx);
        int y_diff = abs(fy-sy);
        int max_diff, min_diff;

        if (x_diff > y_diff){
            max_diff = x_diff;
            min_diff = y_diff;
        }
        else{
            min_diff = x_diff;
            max_diff = y_diff;
        }
        return t>= min_diff + max_diff-min_diff;

    }
};

File: 3056-determine-if-a-cell-is-reachable-at-a-given-time/3056-determine-if-a-cell-is-reachable-at-a-given-time.py
class Solution:
    def isReachableAtTime(self, sx: int, sy: int, fx: int, fy: int, t: int) -> bool:
        if sx == fx and sy == fy and t == 1:
            return False
        x_diff = abs(fx-sx)
        y_diff = abs(fy-sy)
        if x_diff < y_diff:
            min_diff = x_diff
            max_diff = y_diff
        else:
            min_diff = y_diff
            max_diff = x_diff
        return t >= min_diff + max_diff-min_diff